osu.Game.Rulesets.Osu/Mods/OsuModHidden.cs

```diff
@@ -90,5 +110,54 @@ protected override void ApplyHiddenState(DrawableHitObject drawable, ArmedState
                     break;
             }
         }
+
+        protected override void ApplySpecialHiddenState(DrawableHitObject drawable, ArmedState state)
+        {
+            if (!(drawable is DrawableOsuHitObject d))
+                return;
+
+            var h = d.HitObject;
+            getFadeTimes(h);
+
+            switch (drawable)
+            {
+                case DrawableHitCircle circle:
+                    //keep the approach circle visible, only fade out circlepiece
+                    using (drawable.BeginAbsoluteSequence(fadeOutStartTime, true))
+                        circle.CirclePiece.FadeOut(fadeOutDuration);
+
+                    break;
+
+                case DrawableSlider slider:
+                    using (slider.BeginAbsoluteSequence(fadeOutStartTime, true))
+                        slider.Body.FadeOut(longFadeDuration, Easing.Out);
+
+                    break;
+            }
+        }
+
+        private int checkForSpinners(IEnumerable<DrawableHitObject> drawables)
+        {
+            int startingSpinnersCount = 0;
+
+            foreach (var d in drawables.OfType<DrawableOsuHitObject>())
+            {
+                if (d is DrawableSpinner)
+                    ++startingSpinnersCount;
+                else
+                    return startingSpinnersCount;
+            }
+
+            return startingSpinnersCount;
+        }
+
+        private void getFadeTimes(OsuHitObject h)
+        {
+            fadeOutStartTime = h.StartTime - h.TimePreempt + h.TimeFadeIn;
```

Comment:
you can't set fields in a get method and then use outside. needs some rethinking.

Reply:
So it should be something akin to ``AutoProperty``?